lib/ansible/modules/cloud/digital_ocean/digital_ocean_droplet.py

```diff
@@ -0,0 +1,337 @@
+
+class DODroplet(object):
+    def __init__(self, module):
+        self.rest = DigitalOceanHelper(module)
+        self.module = module
+        self.wait = self.module.params.pop('wait', True)
+        self.wait_timeout = self.module.params.pop('wait_timeout', 120)
+        self.unique_name = self.module.params.pop('unique_name', False)
+        # pop the oauth token so we don't include it in the POST data
+        self.module.params.pop('oauth_token')
+
+    def get_by_id(self, droplet_id):
+        if not droplet_id:
+            return None
+        response = self.rest.get('droplets/{0}'.format(droplet_id))
+        json_data = response.json
+        if response.status_code == 200:
+            return json_data
+        return None
+
+    def get_by_name(self, droplet_name):
+        if not droplet_name:
+            return None
+        page = 1
+        while page is not None:
+            response = self.rest.get('droplets?page={0}'.format(page))
+            json_data = response.json
+            if response.status_code == 200:
+                for droplet in json_data['droplets']:
+                    if droplet['name'] == droplet_name:
+                        return {'droplet': droplet}
+                if 'links' in json_data and 'pages' in json_data['links'] and 'next' in json_data['links']['pages']:
+                    page += 1
+                else:
+                    page = None
+        return None
+
+    def get_addresses(self, data):
+        """
+         Expose IP addresses as their own property allowing users extend to additional tasks
+        """
+        _data = data
+        for k, v in data.items():
+            setattr(self, k, v)
+        networks = _data['droplet']['networks']
+        for network in networks.get('v4', []):
+            if network['type'] == 'public':
+                _data['ip_address'] = network['ip_address']
+            else:
+                _data['private_ipv4_address'] = network['ip_address']
+        for network in networks.get('v6', []):
+            if network['type'] == 'public':
+                _data['ipv6_address'] = network['ip_address']
+            else:
+                _data['private_ipv6_address'] = network['ip_address']
+        return _data
+
+    def get_droplet(self):
+        json_data = self.get_by_id(self.module.params['id'])
+        if not json_data and self.unique_name:
+            json_data = self.get_by_name(self.module.params['name'])
+        return json_data
+
+    def create(self):
+        json_data = self.get_droplet()
+        droplet_data = None
+        if json_data:
+            droplet_data = self.get_addresses(json_data)
+            self.module.exit_json(changed=False, data=droplet_data)
+        if self.module.check_mode:
+            self.module.exit_json(changed=True)
+        response = self.rest.post('droplets', data=self.module.params)
+        json_data = response.json
+        self.module.fail_json(changed=False, msg=response.status_code)
```

Comment:
This line is causing the module to fail unconditionally on every create.

Reply:
Looks like we just need this line removed and we should be good to merge.